### PR TITLE
day9

### PR DESCRIPTION
**Pull request by marvin-hsu**

### CODE REVIEW

File: 2023/09/marvinhsu/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    let first = layer[0];
+    match get_sub_layer(layer) {
+        Some(sub_layer) => first - get_previous_number(sub_layer.as_slice()),
+        None => first,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::vec;
```
Comment (weihanglo) [inline 2023/09/marvinhsu/src/lib.rs:60]: This is not necessary. vec is in prelude https://doc.rust-lang.org/std/prelude/index.html

File: 2023/09/marvinhsu/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    #[test]
+    fn test_part1() {
+        let input = include_str!("../example");
+        assert_eq!(part1(input).unwrap(), 114);
+    }
+
+    #[test]
+    fn test_get_sub_layer() {
+        assert_eq!(
+            get_sub_layer(&vec![1, 3, 6, 10, 15, 21]).unwrap(),
```
Comment (weihanglo) [inline 2023/09/marvinhsu/src/lib.rs:73]: This should work without extra vec allocation and more idiomatic.

```suggestion
            get_sub_layer(&[1, 3, 6, 10, 15, 21]).unwrap(),
```

**Review by weihanglo — APPROVED** [review]
Let me know if you want to update or proceed to merge.

**Comment by marvin-hsu** [comment]
> Let me know if you want to update or proceed to merge.

Already removed redundancy. Thank you very much!

**Pull request merged**